meetings/2020/WASI-10-22.md

```diff
@@ -44,3 +44,161 @@ Installation is required, see the calendar invite.
+
+SC: Unanimous consent achieved to move to stage 2.
+
+AB: I will submit a PR updating which table it is in.
+
+#### Character encodings
+
+DG: Several threads about encodings going on. Want feedback about direction.
+
+<DG presenting> TODO: Add link to slides
+
+SC: Won’t the C program truncate on strcpy?
+
+DG: Yes, but then you’ll get a file not found error. <Back to slides>
+
+SC: Where does this ARF encoding come from?
+
+DG: I invented this after a lot of tries.
+
+SA: Is this for filenames only or also for contents?
+
+DG: File names, env vars, commandline args, but not file contents.
+
+SC: Anywhere a known string crosses the WASI boundary?
+
+DG: There might be other places. Where invalid strings should be roundtrippable without trapping. Some APIs trap, others wouldn’t. Some tools will want different things as well.
+
+TL: … userspace virtualization… ?
+
+PH: When IT describe interfaces, it would be great to use strings for these types, and IT guarantees unicode. ARF lets us use the IT string type.
+
+AC: In windows, filenames are lists of 16-bit values. How do we go from 16-bit to 8-bit strings? We need an extra layer.
+
+DG: We can extend ARF to handle Windows character space as well. ARF would be a little different on Windows and Linux.
+
+SC: missed this
+
+DG: wasi engines can translate host character encodings into unicode.
+
+AC: What would the users look like? Not raw API users, but e.g. rust stdlib users. Would the entire ARF string be stored in Rust’s Path object?
+
+DG: Two cases: in do nothing case, use all the standard APIs. In ARF encoding case, get string as ARF encoded OSString or similar. WASI will either detect ARF string and not validate or would get file not found. There is a Rust library that can take an OSString and be ARF-aware to do things.
+
+SC: But most programs won’t need to do this because these files don’t exist in practice.
+
+AC: Not sure about this. Rust stdlib would want to do the proper decoding to show better error messages. It seems only extremely low-level C developers would be in the do-nothing case.
+
+SC: Do-nothing only prints first half, right?
+
+AC: Only in C. When there is a pointer and length, the second half gets printed too.
+
+DG: Good point. Don’t have an answer for that.
+
+SC: Curious about prior art.
+
+DG: Perl 6 has UTF8-C8. Uses highest private use code point as an escape char.
+
+AC: Using nul character seems reasonable given that native filesystems already disallow it.
+
+DG: Python uses lone surrogates. Not great because you end up with invalid unicode.
+
+SC: To be clear, status quo is passing through raw bytes from OS with no validation?
+
+DG: Some implementations already have some validation and different behavior.
+
+#### Discuss about how to move forward with a WASI logo
+
+SA: I’m the founder and CEO of Wasmer, want to see what the risks are about
+moving forward with a different logo and what are the issues with the current
+logo.
+
+<SA presenting> TODO: add slides
```

Comment:
@syrusakbary When you get a chance, could you upload your slides and add a link to them?